core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java

```diff
@@ -113,27 +113,48 @@
 	private String correlationIdToHexPrefix = "ID:";
 	private int correlationIdMaxLength = -1;
 
-	public static int stringToDeliveryMode(String mode) {
-		if (MODE_PERSISTENT.equalsIgnoreCase(mode)) {
-			return DeliveryMode.PERSISTENT;
-		}
-		if (MODE_NON_PERSISTENT.equalsIgnoreCase(mode)) {
-			return DeliveryMode.NON_PERSISTENT;
+	public enum AcknowledgeModeEnum {
+		NOT_SET(0, ""),
+		AUTO_ACKNOWLEDGE(Session.AUTO_ACKNOWLEDGE, "auto"),
+		CLIENT_ACKNOWLEDGE(Session.CLIENT_ACKNOWLEDGE, "client"),
+		DUPS_OK_ACKNOWLEDGE(Session.DUPS_OK_ACKNOWLEDGE, "dups");
+		
+		int acknowledgeMode;
+		String shortName;
```

Comment:
Het is veiliger als dit private fields zijn. Ze mogen nooit aangepast worden en dat zou nu eventueel wel kunnen.